src/core/rotation.cpp

```diff
@@ -146,56 +146,56 @@ void propagate_omega_quat_particle(Particle &p, double time_step) {
   assert(square >= 0.);
   auto const lambda = 1 - S[0] * 0.5 * time_step_squared - sqrt(square);
 
-  p.m.omega += time_step_half * Wd;
-  p.r.quat += time_step * (Qd + time_step_half * Qdd) - lambda * p.r.quat;
+  p.omega() += time_step_half * Wd;
+  p.quat() += time_step * (Qd + time_step_half * Qdd) - lambda * p.quat();
 
   /* and rescale quaternion, so it is exactly of unit length */
-  auto const scale = p.r.quat.norm();
+  auto const scale = p.quat().norm();
   if (scale == 0) {
-    p.r.quat = Utils::Quaternion<double>::identity();
+    p.quat() = Utils::Quaternion<double>::identity();
   } else {
-    p.r.quat /= scale;
+    p.quat() /= scale;
```

Comment:
p.quat().normalize() ?

Reply:
Same issue as in the integrator, only worse: the quaternion normalize method uses `boost::qvm` under the hood, therefore that single C++ line, which takes 50 lines of assembly code, becomes 700 lines of assembly code with `p.quat().normalize()`.